src/robotFactory.js

```diff
@@ -1,10 +1,78 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, position, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = position.x === undefined
+      ? {
+        x: 0, y: 0,
+      }
+      : position;
+    this.chipVersion = chipVersion;
+  }
 
-class DeliveryDrone {}
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  getInfo() {
+    return `Robot: ${this.name}, Chip version: ${
+      this.chipVersion}, Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, position, chipVersion) {
+    super(name, weight, position, chipVersion);
+
+    this.coords = position.z === undefined
+      ? {
+        z: 0,
+      }
+      : position;
+  }
+
+  goUp(step = 1) {
+    this.coords.z += step;
+  }
+  goDown(step = 1) {
+    this.coords.z -= step;
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(name, weight, position, chipVersion, maxLoadWeight, currentLoad) {
```

Comment:
multi-line properties if there are more than 4 of them:
```suggestion
  constructor(
    name,
    weight,
    position,
    chipVersion,
    maxLoadWeight,
    currentLoad
  ) {
```